src/helpers.rs

```diff
@@ -912,30 +912,57 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         })
     }
 
-    /// Read a sequence of bytes until the first null terminator.
-    fn read_c_str<'a>(&'a self, ptr: Pointer<Option<Provenance>>) -> InterpResult<'tcx, &'a [u8]>
```

Comment:
`read_c_str` is different than the others, since it can directly return `&'a [u8]`. It probably shouldn't be part of the unified `read_any_str` function.

Reply:
Isn't the same true for `read_wide_str`? My new implementation returns `&'a [u16]` instead of the previous `Vec<u16>`.

Reply:
It's not possible to return `&'a [u16]`, I think. Due to endianess the data we return may not match what exists in memory.

Reply:
Makes sense. Do we have any big endian miri targets, so that we can test that the miri tests also works on those?

Reply:
Yeah, CI runs the tests on a big-endian target.

Reply:
Hmm, but then the test succeeds with returning `&[u16]`. So does that mean the tests aren't good enough or that it is ok to return `&[u16]`?